app/src/main/java/org/supla/android/Preferences.java

```diff
@@ -105,7 +105,8 @@ private byte[] getRandom(String pref_key, int size) {
             if (_prefs.getBoolean(pref_key + "_encrypted", false)) {
                 result = Encryption.decryptDataWithNullOnException(result, getDeviceID(), true);
             }
-            encryptAndSave(pref_key, result);
```

Comment:
Please change to:

if(result != null) {
         encryptAndSave(pref_key, result);
}